src/main/java/io/kabanero/api/GitHubClientInitilizer.java

```diff
@@ -0,0 +1,41 @@
+package io.kabanero.api;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.egit.github.core.client.GitHubClient;
+
+import io.kabanero.v1alpha2.client.apis.KabaneroApi;
+import io.kabanero.v1alpha2.models.Kabanero;
+import io.kubernetes.KabaneroClient;
+import io.kubernetes.client.ApiClient;
+
+public class GitHubClientInitilizer {
+    private final static Logger LOGGER = Logger.getLogger(GitHubClientInitilizer.class.getName());
+
+    private final static String DEFAULT_NAMESPACE = "kabanero";
+
+    public static GitHubClient getClient(String instanceName) throws IOException, GeneralSecurityException {
+        String apiUrl = getApiURL(instanceName);
+        if("https://api.github.com".equals(apiUrl)){
+            return new GitHubClient(); 
+        }else{
+            return new GitHubClient(apiUrl);            
+        }
+    }
```

Comment:
any reason this can't be: `return new GitHubClient(apiUrl);` ? seems like it would initialize it correct either way and save us the extra lines from an `if` statement

Reply:
if that's possible I would also check to see what happens then a bad url is put into GitHubClient like `return new GitHubClient("https://kfjdklfjdsklfjs.com); ` and add proper error handling for it if it chokes.